spec/models/daily_report_spec.rb

```diff
@@ -15,43 +15,50 @@
         after(:each) { Time.zone = default_zone }
 
         it "should work on a regular day" do
-          time = Time.zone.local(2017, 1, 2, 10, 0) # 2017-01-02 10:00
+          time = Time.zone.local(2017, 1, 2, 10, 0)
           day = time.beginning_of_day
-          shown_posts = Array.new(24) do |i| # 0 .. 23
-            step = day + i.hours
-            Timecop.freeze(step) { create(:post) }
-          end
+          shown_posts = make_shown_posts(time)
           shown_posts.each do |post|
             expect(post.tagged_at).to be_between(day, day.end_of_day).inclusive
           end
 
-          hidden_post1 = Timecop.freeze(day - 1.hour) { create(:post) }
-          hidden_post2 = Timecop.freeze(day.end_of_day + 1.hour) { create(:post) }
-          expect(hidden_post1.tagged_at).not_to be_between(day, day.end_of_day).inclusive
-          expect(hidden_post2.tagged_at).not_to be_between(day, day.end_of_day).inclusive
+          hidden_posts = make_hidden_posts(time)
+          hidden_posts.each do |post|
+            expect(post.tagged_at).not_to be_between(day, day.end_of_day).inclusive
+          end
 
           expect(DailyReport.new(time).posts).to match_array(shown_posts)
         end
 
         it "should work on a daylight change day" do
           time = Time.zone.local(*dst_day_params)
-          # clock goes back; 25 hours in the day
           day = time.beginning_of_day
-          shown_posts = Array.new(25) do |i| # 0 .. 24
-            step = day + i.hours
-            Timecop.freeze(step) { create(:post) }
-          end
+          # clock goes back; 25 hours in the day
+          shown_posts = make_shown_posts(time)
```

Comment:
It looks like this is only using 24 posts rather than 25 now. Maybe this should be a parameter for `make_shown_posts`? (The reason to do 25 is to check this works around DST fall-back time transitions)